pkg/booleanpolicy/policyversion/upgrade.go

```diff
@@ -1,643 +1,67 @@
-			{
-				Value: fmt.Sprintf("%s=%s", p.GetKey(), p.GetValue()),
-			},
-		},
-	}
-}
-
-func convertVolumePolicy(fields *storage.PolicyFields) []*storage.PolicyGroup {
-	p := fields.GetVolumePolicy()
-	if p == nil {
-		return nil
-	}
-
-	var res []*storage.PolicyGroup
-	if p.GetName() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.VolumeName,
-			Values:    getPolicyValues(p.GetName()),
-		})
-	}
-
-	if p.GetType() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.VolumeType,
-			Values:    getPolicyValues(p.GetType()),
-		})
-	}
-
-	if p.GetDestination() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.VolumeDestination,
-			Values:    getPolicyValues(p.GetDestination()),
-		})
-	}
-
-	if p.GetSource() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.VolumeSource,
-			Values:    getPolicyValues(p.GetSource()),
-		})
-	}
-
-	ro := p.GetSetReadOnly()
-	if ro != nil {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.WritableMountedVolume,
-			Values:    getPolicyValues(!p.GetReadOnly()),
-		})
-	}
-
-	return res
-}
-
-func convertPortPolicy(fields *storage.PolicyFields) []*storage.PolicyGroup {
-	p := fields.GetPortPolicy()
-	if p == nil {
-		return nil
-	}
-
-	var res []*storage.PolicyGroup
-	if p.GetPort() != 0 {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ExposedPort,
-			Values:    getPolicyValues(int64(p.GetPort())),
-		})
-	}
-
-	if p.GetProtocol() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ExposedPortProtocol,
-			Values:    getPolicyValues(p.GetProtocol()),
-		})
-	}
-
-	return res
-}
-
-func convertProcessPolicy(fields *storage.PolicyFields) []*storage.PolicyGroup {
-	p := fields.GetProcessPolicy()
-	if p == nil {
-		return nil
-	}
-
-	var res []*storage.PolicyGroup
-	if p.GetName() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ProcessName,
-			Values:    getPolicyValues(p.GetName()),
-		})
-	}
-
-	if p.GetAncestor() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ProcessAncestor,
-			Values:    getPolicyValues(p.GetAncestor()),
-		})
-	}
-
-	if p.GetArgs() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ProcessArguments,
-			Values:    getPolicyValues(p.GetArgs()),
-		})
-	}
-
-	if p.GetUid() != "" {
-		res = append(res, &storage.PolicyGroup{
-			FieldName: fieldnames.ProcessUID,
-			Values:    getPolicyValues(p.GetUid()),
-		})
-	}
-
-	return res
-}
-
-func convertHostMountPolicy(fields *storage.PolicyFields) []*storage.PolicyGroup {
-	p := fields.GetHostMountPolicy()
-	if p.GetSetReadOnly() == nil {
-		return nil
-	}
-
-	return []*storage.PolicyGroup{{
-		FieldName: fieldnames.WritableHostMount,
-		Values:    getPolicyValues(!p.GetReadOnly()),
-	},
-	}
-}
-
-const (
-	capPrefix = "CAP_"
+// Upgrader takes a policy in version N and *must* convert it to N+1.
```

Comment:
This file has not been deleted! Hopefully this comment will force GitHub to render it.

It's been massively trimmed down by deleting anything that hd to do with fields or old versions. It simply has the same interface as downgrade.go now. But it doesn't do anything since there's only one version (1.1) that's active.